src/SOS/Strike/sosextensions.cpp

```diff
@@ -0,0 +1,109 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "exts.h"
+#ifndef FEATURE_PAL
+#include "dbgengservices.h"
+#endif
+
+SOSExtensions::SOSExtensions(IDebuggerServices* debuggerServices, IHost* host) :
+    Extensions(debuggerServices)
+{
+    m_pHost = host;
+    OnUnloadTask::Register(SOSExtensions::Uninitialize);
+}
+
+#ifndef FEATURE_PAL
+SOSExtensions::~SOSExtensions()
+{
+    if (m_pDebuggerServices != nullptr)
+    {
+        ((DbgEngServices*)m_pDebuggerServices)->Uninitialize();
+        m_pDebuggerServices->Release();
+        m_pDebuggerServices = nullptr;
+    }
+}
+#endif
+
+#ifndef FEATURE_PAL
```

Comment:
Why are these two lines here? They could just be grouped together under the #ifndef FEATURE_PAL.

Reply:
I'll make one conditional block.